patroni/api.py

```diff
@@ -175,13 +175,29 @@ def do_POST_failover(self):
         member = request.get('member', None)
         cluster = self.server.patroni.ha.dcs.get_cluster()
         status_code = 503
-        data = self.is_failover_possible(cluster, leader, member)
-        if not data:
-            if not self.server.patroni.dcs.manual_failover(leader, member):
-                data = b'failed to write failover key into DCS'
-            else:
-                self.server.patroni.dcs.event.set()
-                status_code, data = self.poll_failover_result(cluster.leader and cluster.leader.name, member)
+
+        data = b''
+        if request.get('planned_at'):
+            try:
+                planned_at = parse_datetime(request['planned_at'])
+                planned_at = localize_datetime(planned_at)
+                if planned_at < utcnow_timezone_aware():
+                    data = b'Cannot schedule failover in the past'
+                    status_code = 422
+                elif self.server.patroni.dcs.manual_failover(leader, member, planned_at):
+                    data = b'Failover scheduled'
+                    status_code = 200
+            except (ValueError, TypeError):
+                logger.exception('Invalid planned failover time: {}'.format(request['planned_at']))
+                data = b'Unable to parse planned timestamp. Should be in ISO 8601 format'
```

Comment:
I think we should always use the ISO 8601 in the API, but provide a more user-friendly interface for patronictl. I don't think calling the parse_datetime and localize_datetime everywhere just in case in a sane way to go, it doesn't make the code more readable and it's easy to forget it somewhere. Instead, what about calling it only in patronictl, and just enforcing the ISO 8601 format with strptime in the API endpoint?